File: sol/2044/2044.py
from collections import Counter

class Solution:
    def countMaxOrSubsets(self, nums: List[int]) -> int:
        c = Counter()
        c[0] = 1
        target = 0
        for x in nums:
            target |= x
            newC = Counter(c)
            for k, v in c.items():
                newC[k | x] += v
            c = newC
        return c[target]

File: sol/2044/2044.cpp
#include <unordered_map>

struct Solution {
    int countMaxOrSubsets(vector<int>& nums) {
        unordered_map<int, int> c;
        c[0] = 1;
        int target = 0;
        for (int x : nums){
            target |= x;
            unordered_map<int, int> newc(c);
            for (auto [k, v] : c){
                newc[k | x] += v;
            }
            swap(c, newc);
        }
        return c[target];
    }
};
